test/tests/imageserver/test_image_service_system.py

```diff
@@ -0,0 +1,319 @@
+                fit_common.time.sleep(30)
+            elif result['json']["status"] == 'succeeded':
+                logs.debug("OS Install workflow state: {}".format(result['json']["status"]))
+                return True
+            else:
+                break
+        logs.error("Task failed with the following state: " + result['json']["status"])
+        return False
+
+    def _get_tester_ip(self):
+        serverip = self._get_serverip()
+        monip = FILE_CONFIG["rackhd_control_ip"]
+        cmd = "ping -R -c 1 " + monip + ""
+        (command_output, exitstatus) = pexpect.run(
+            "ssh -q -o StrictHostKeyChecking=no -t " + FILE_CONFIG['usr'] + "@" + serverip +
+            " sudo bash -c \\\"" + cmd + "\\\"", withexitstatus=1,
+            events={"assword": FILE_CONFIG['pwd'] + "\n"}, timeout=300)
+        uud = command_output.split("\t")
+        myip = uud[1].split("\r\n")[0]
+        logs.debug('My IP address is: ' + myip)
+        return myip
+
+    def _create_esxi_repo(self):
+        logs.debug("create a ESXi repo")
+        for osrepo in FILE_CONFIG["os_image"]:
+            if osrepo["osname"] == "ESXi" and osrepo["version"] == "6.0":
+                os_name = osrepo["osname"]
+                os_version = osrepo["version"]
+                http_iso_url = osrepo["url"]
+                self._upload_os_by_network(os_name, os_version, http_iso_url)
+                break
+
+    def _delete_all_images(self):
+        os_image_list = self._list_os_image()
+        serverip = self._get_serverip()
+        for image_repo in os_image_list:
+            self.assertNotEqual(
+                self._delete_os_image(image_repo["name"], image_repo["version"]), "fail", "delete image failed!")
+            fileurlprefix = "http://" + serverip + ":9090/" + image_repo["name"] + '/' + image_repo["version"] + '/'
+            self.assertFalse(self._file_exists(fileurlprefix), "The repo url does not deleted completely")
+        os_image_list_clear = self._list_os_image()
+        self.assertTrue(os_image_list_clear == [])
+        os_iso_list = self._list_os_iso()
+        for iso_repo in os_iso_list:
+            self.assertNotEqual(self._delete_os_iso(iso_repo["name"]), "fail", "delete iso failed!")
+        os_iso_list_clear = self._list_os_iso()
+        self.assertTrue(os_iso_list_clear == [], "The iso does not deleted completely")
+        logs.debug("All repo is cleared!")
+
+    def _upload_microkernel(self, filename):
+        myfile = open(filename, 'rb')
+        serverip = self._get_serverip()
+        mon_url = '/microkernel?name=' + filename
+        response = fit_common.restful("http://" + serverip + ":7070" + mon_url, rest_action="binary-put",
+                                      rest_payload=myfile)
+        if response['status'] in range(200, 205):
+            return response['json']
+        else:
+            logs.debug_3('Incorrect HTTP return code, expected 201, got:' + str(response['status']))

```

Comment:
expected 201, looks like you expect anything in 201 to 205 here?  same comment thru the rest of the file.